#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_wsdl2_duplicate_keys_r117243.dpatch by Jo Shields <directhex@apebox.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix glitch in CLI 2.0 version of System.Web.Services which can
## DP: cause wsdl files to be badly processed, due to key duplication

@DPATCH@
Index: mcs/class/System.Web.Services/System.Web.Services.Description/BasicProfileChecker.cs
===================================================================
--- mcs/class/System.Web.Services/System.Web.Services.Description/BasicProfileChecker.cs	(revision 117242)
+++ mcs/class/System.Web.Services/System.Web.Services.Description/BasicProfileChecker.cs	(revision 117243)
@@ -177,7 +177,7 @@
 				foreach (OperationMessage om in op.Messages) {
 					Message msg = ctx.Services.GetMessage (om.Message);
 					foreach (MessagePart part in msg.Parts)
-						parts.Add (part,part);
+						parts [part] = part; // do not use Add() - there could be the same MessagePart instance.
 				}
 			}
 			
